R/model_builder.R

```diff
@@ -201,8 +192,8 @@ build_kmeans.cols <- function(df, ...,
     if(augment){
       ret <- broom::augment(kmeans_ret, df)
       colnames(ret)[[ncol(ret)]] <- avoid_conflict(grouped_column, "cluster")
-      # cluster column is factor labeled "1", "2"..., so convert it to integer to avoid confusion
-      ret[[ncol(ret)]] <- as.integer(ret[[ncol(ret)]])
+      # make the clustering labels factor with levels with sequence of numbers
+      ret[[ncol(ret)]] <- factor(ret[[ncol(ret)]], levels=seq(centers))
```

Comment:
Isn't ret[[ncol(ret)]] already a factor according to the code before change?
If so do we need to apply factor() again?